Remove MSW wxUxThemeEngine class

TcT2k opened the pull request:
This undocumented "private" class was used for various windows UxTheme
functions which are available since WinXP. As wxWidgets 3.1 is XP+ it
does not make sense anymore to load the theme functions dynamically.

TcT2k committed:
Remove MSW wxUxThemeEngine class

This undocumented "private" class was used for various windows UxTheme
functions which are available since WinXP. As wxWidgets 3.1 is XP+ it
does not make sense anymore to load the theme functions dynamically.

vadz commented:
Thanks! I agree that this would be nice to clean up, even if it's not the most urgent thing to do, but the AppVeyor has found a problem with the MinGW build: apparently we don't need to redefine all this stuff now that we include `uxtheme.h` directly. Or maybe only if e.g. `TMT_MENUFONT` is not already defined if it's missing in any still supported MinGW versions. Looking at the versions here, I see these symbols in 4.8.0 but not the previous versions. I don't know if there are still people using anything earlier than that...

TcT2k committed:
Remove unused definitions in msw/checkbox.cpp

TcT2k committed:
Move UXTheme symbol definitions for Vista+ to msw/uxtheme.h

Remove various definitions and symbol declarations from
numerous files using msw/uxtheme.h to a single file.
When possible vssym32.h is included from the Windows SDK.
For older SDKs tmschema.h is included and missing symbols
are defined in msw/uxtheme.h.

TcT2k commented:
@vadz I've made some changes which should allow this to build with any WinXP compatible SDK (which should include older Visual Studio Versions and MinGW).
All definitions regarding theme support are now in `msw/uxtheme.h` instead of the individual files where themes are used for drawing. This gets rid of some duplicate definitions and allows for easy removal when support for building with SDKs older than Vista gets dropped in the future.

vadz commented:
Thanks, I'm merging it with only cursory review, but I guess we'll find out about any problems soon enough :-)

Thanks again for taking time to clean this up!
